### PR TITLE
Add test_csi.py::test_csi_encrypted_migratable_block_volume

### PR DESCRIPTION
**Pull request by derekbit**
- Add test_csi.py::test_csi_encrypted_migratable_block_volume
- Fix test_csi.py::test_csi_encrypted_block_volume

Longhorn/longhorn#7678

### CODE REVIEW

**Comment by derekbit** [comment]
```
platform linux -- Python 3.9.18, pytest-5.3.1, py-1.11.0, pluggy-0.13.1 -- /usr/bin/python3.9
cachedir: .pytest_cache
rootdir: /integration, inifile: pytest.ini
plugins: order-1.0.1, repeat-0.9.1
collected 15 items

test_csi.py::test_csi_mount PASSED
test_csi.py::test_csi_io PASSED
test_csi.py::test_csi_backup[s3] PASSED
test_csi.py::test_csi_backup[nfs] PASSED
test_csi.py::test_csi_block_volume PASSED
test_csi.py::test_csi_encrypted_block_volume PASSED
test_csi.py::test_csi_encrypted_migratable_block_volume PASSED
test_csi.py::test_csi_offline_expansion PASSED
test_csi.py::test_csi_block_volume_online_expansion 2500+0 records in
2500+0 records out
2621440000 bytes (2.6 GB, 2.4 GiB) copied, 13.1696 s, 199 MB/s
PASSED
test_csi.py::test_csi_mount_volume_online_expansion 2500+0 records in
2500+0 records out
2621440000 bytes (2.4GB) copied, 9.065344 seconds, 275.8MB/s
PASSED
test_csi.py::test_xfs_pv PASSED
test_csi.py::test_xfs_pv_existing_volume meta-data=/dev/longhorn/longhorn-testvol-5esmx9 isize=512    agcount=1, agsize=4096 blks
         =                       sectsz=512   attr=2, projid32bit=1
         =                       crc=1        finobt=1, sparse=0, rmapbt=0
         =                       reflink=0
data     =                       bsize=4096   blocks=4096, imaxpct=25
         =                       sunit=0      swidth=0 blks
naming   =version 2              bsize=4096   ascii-ci=0, ftype=1
log      =internal log           bsize=4096   blocks=855, version=2
         =                       sectsz=512   sunit=0 blks, lazy-count=1
realtime =none                   extsz=4096   blocks=0, rtextents=0
PASSED
test_csi.py::test_csi_expansion_with_replica_failure PASSED
test_csi.py::test_allow_volume_creation_with_degraded_availability_csi PASSED
test_csi.py::test_csi_minimal_volume_size PASSED
```

**Review by yangchiu — DISMISSED** [review]
LGTM

**Comment by derekbit** [comment]
cc @c3y1huang Can you help review it? Thanks.

File: manager/integration/tests/common.py [hunk 1]
```diff
@@ -6111,3 +6111,13 @@ def wait_for_instance_manager_count(client, number, retry_counts=120):
...
+
+
+def wait_delete_dm_device(api, name):
+    for i in range(RETRY_COUNTS):
+        path = os.path.join("/dev/mapper" + name)
+        found = os.path.exists(path)
+        if not found:
+            break
+        time.sleep(RETRY_INTERVAL)
+    assert not found
```
Comment (c3y1huang) [inline manager/integration/tests/common.py:6123]: Suggest include an error message for better error reporting.

File: manager/integration/tests/common.py [hunk 2]
```diff
@@ -6111,3 +6111,13 @@ def wait_for_instance_manager_count(client, number, retry_counts=120):
         time.sleep(RETRY_INTERVAL_LONG)
 
     return len(ims)
+
+
+def wait_delete_dm_device(api, name):
+    for i in range(RETRY_COUNTS):
+        path = os.path.join("/dev/mapper" + name)
```
Comment (c3y1huang) [inline manager/integration/tests/common.py:6118]:
```suggestion
        path = os.path.join("/dev/mapper", name)
```

Comment (c3y1huang) [reply manager/integration/tests/common.py:6118]: And suggest move this outside of the loop.

File: manager/integration/tests/common.py [hunk 3]
```diff
@@ -6111,3 +6111,13 @@ def wait_for_instance_manager_count(client, number, retry_counts=120):
         time.sleep(RETRY_INTERVAL_LONG)
 
     return len(ims)
+
+
+def wait_delete_dm_device(api, name):
```
Comment (c3y1huang) [inline manager/integration/tests/common.py:6116]: Where is `api` used?

File: manager/integration/tests/test_csi.py [hunk 4]
```diff
@@ -290,20 +291,55 @@ def test_csi_encrypted_block_volume(client, core_api, storage_class, crypto_secr
...
     create_and_verify_block_volume(client, core_api, storage_class, pvc,
-                                   pod_manifest)
+                                   pod_manifest, False)
 
 
-def create_and_verify_block_volume(client, core_api, storage_class, pvc, pod_manifest):  # NOQA
+@pytest.mark.csi  # NOQA
+def test_csi_encrypted_migratable_block_volume(client, core_api, storage_class, crypto_secret, pvc, pod_manifest):  # NOQA
+    """
+    Test CSI feature: encrypted migratable block volume
```
Comment (c3y1huang) [inline manager/integration/tests/test_csi.py:304]: Suggest add an issue link for context referencing.

**Review by c3y1huang — CHANGES_REQUESTED** [review]

**Comment by innobead** [comment]
@derekbit let's move this PR forward.

File: manager/integration/tests/common.py (hunk 3 above)
Comment (derekbit) [reply manager/integration/tests/common.py:6116]: Used by `create_and_verify_block_volume`

File: manager/integration/tests/common.py (hunk 1 above)
Comment (derekbit) [reply manager/integration/tests/common.py:6123]: Could you elaborate more on or add an example for the comment? The implementation follows. the convention of similar functions in common.py. The traceback log message might be enough for debugging.

Comment (derekbit) [reply manager/integration/tests/common.py:6123]: cc @c3y1huang

File: manager/integration/tests/test_csi.py (hunk 4 above)
Comment (derekbit) [reply manager/integration/tests/test_csi.py]: Added. Thanks.

File: manager/integration/tests/common.py (hunk 2 above)
Comment (derekbit) [reply manager/integration/tests/common.py]: Updated. Thank you.

File: manager/integration/tests/common.py (hunk 1 above)
Comment (c3y1huang) [reply manager/integration/tests/common.py:6163]: `assert not found` doesn't output what is not found when it fails, so suggest adding a message, for example:
```
assert not found, f"Failed to delete {path}"
```

**Review by c3y1huang — APPROVED** [review]

**Comment by derekbit** [comment]
```
longhorn-test:/integration/tests # pytest -vs test_csi.py::test_csi_encrypted_migratable_block_volume
============================================== test session starts ===============================================
platform linux -- Python 3.11.9, pytest-6.2.4, py-1.11.0, pluggy-0.13.1 -- /usr/bin/python3.11
cachedir: .pytest_cache
rootdir: /integration, configfile: pytest.ini
plugins: order-1.0.1, repeat-0.9.1
collected 1 item

test_csi.py::test_csi_encrypted_migratable_block_volume PASSED
```

**Comment by derekbit** [comment]
@khushboo-rancher Updated. Thank you.

**Pull request merged**